crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -148,24 +199,24 @@ mod orderbook {
 
         fn create_order(ref self: ContractState, order: OrderV1, sign_info: SignInfo) {
             let block_ts = starknet::get_block_timestamp();
-            order.validate_common_data(block_ts).expect(orderbook_errors::ORDER_INVALID_DATA);
+            let validation = order.validate_common_data(block_ts);
+            if validation.is_err() {
+                panic_with_felt252(validation.unwrap_err().into());
```

Comment:
@kwiss @glihm I'm curious, what the diff between`panic()` and `assert()` to add condition and returns errors if the condition failed? And what's considered best practice?